tests/pytests/unit/utils/parsers/test_log_parsers.py

```diff
@@ -0,0 +1,775 @@
+
+
+@pytest.fixture
+def parser(log_cli_parser):
+    param_map = {
+        "master": salt.utils.parsers.MasterOptionParser,
+        "minion": salt.utils.parsers.MinionOptionParser,
+        "proxyminion": salt.utils.parsers.ProxyMinionOptionParser,
+        "syndic": salt.utils.parsers.SyndicOptionParser,
+        "saltcmd": salt.utils.parsers.SaltCMDOptionParser,
+        "saltcp": salt.utils.parsers.SaltCPOptionParser,
+        "saltkey": salt.utils.parsers.SaltKeyOptionParser,
+        "saltcall": salt.utils.parsers.SaltCallOptionParser,
+        "saltrun": salt.utils.parsers.SaltRunOptionParser,
+        "saltssh": salt.utils.parsers.SaltSSHOptionParser,
+        "saltcloud": salt.utils.parsers.SaltCloudParser,
+        "spm": salt.utils.parsers.SPMParser,
+        "saltapi": salt.utils.parsers.SaltAPIParser,
+    }
+    return param_map[log_cli_parser]
+
+
+@pytest.fixture
+def config_func(log_cli_parser):
+    param_map = {
+        "master": "salt.config.master_config",
+        "minion": "salt.config.minion_config",
+        "proxyminion": "salt.config.proxy_config",
+        "syndic": "salt.config.syndic_config",
+        "saltcmd": "salt.config.client_config",
+        "saltcp": "salt.config.master_config",
+        "saltkey": "salt.config.client_config",
+        "saltcall": "salt.config.minion_config",
+        "saltrun": "salt.config.master_config",
+        "saltssh": "salt.config.master_config",
+        "saltcloud": "salt.config.cloud_config",
+        "spm": "salt.config.spm_config",
+        "saltapi": "salt.config.api_config",
+    }
+    return param_map[log_cli_parser]
+
+
+@pytest.fixture
+def log_file(tmp_path, logfile_config_setting_name):
+    return str(tmp_path / logfile_config_setting_name)
+
+
+@pytest.fixture
+def args(log_cli_parser):
+    if log_cli_parser in ("saltcmd", "saltssh"):
+        return ["foo", "bar.baz"]
+    elif log_cli_parser == "saltcp":
+        return ["foo", "bar", "baz"]
+    elif log_cli_parser in ("saltcall", "saltrun"):
+        return ["foo.bar"]
+    elif log_cli_parser == "saltcloud":
+        return ["-p", "foo", "bar"]
+    elif log_cli_parser == "spm":
+        return ["foo", "bar"]
+    return []
+
+
+@pytest.fixture
+def loglevel_config_setting_name():
+    return "log_level"
+
+
+@pytest.fixture
+def logfile_config_setting_name(log_cli_parser):
+    if log_cli_parser == "syndic":
+        return "syndic_log_file"
+    elif log_cli_parser == "saltkey":
+        return "key_logfile"
+    elif log_cli_parser == "saltssh":
+        return "ssh_log_file"
+    elif log_cli_parser == "spm":
+        return "spm_logfile"
+    elif log_cli_parser == "saltapi":
+        return "api_logfile"
+    return "log_file"
+
+
+@pytest.fixture
+def logfile_loglevel_config_setting_name():
+    return "log_level_logfile"
+
+
+@pytest.fixture
+def testing_config(default_config, root_dir, logfile_config_setting_name, log_file):
+    _testing_config = default_config.copy()
```

Comment:
```suggestion
```
It's already a copy, and the scope of the fixture is function, so it gets recreated for every test, no need to copy it again.

Reply:
We use the default config as a seperate fixture in one of the tests, so I left the copy in there.

Reply:
hmm, ok.